bundles/model/org.eclipse.smarthome.model.rule.runtime/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleTriggerManager.java

```diff
@@ -273,12 +279,42 @@ public RuleTriggerManager(Injector injector) {
         }
     }
 
-    private Iterable<Rule> internalGetRules(TriggerTypes triggerType, Item item, Type oldType, Type newType) {
-        List<Rule> result = new ArrayList<>();
-        Iterable<Rule> rules = getAllRules(triggerType, item.getName());
+    private Iterable<Rule> getRulesOrEmptySet(TriggerTypes type, String name) {
```

Comment:
IMHO it would be better to amend `getAllRules()` accordingly. There were only two usages of it - and both guard themselves anyway by creating an empty collection if there was null returned.